saleor/account/models.py

```diff
@@ -182,6 +186,57 @@ def get_ajax_label(self):
         return self.email
 
 
+class ServiceAccount(ModelWithMetadata):
+    name = models.CharField(max_length=60)
+    auth_token = models.CharField(default=generate_token, unique=True, max_length=30)
+    created = models.DateTimeField(auto_now_add=True)
+    is_active = models.BooleanField(default=True)
+    permissions = models.ManyToManyField(
+        Permission,
+        verbose_name=_("service account permissions"),
+        blank=True,
+        help_text=_("Specific permissions for this service."),
+        related_name="service_set",
+        related_query_name="service",
+    )
+
+    class Meta:
+        permissions = (
+            (
+                "manage_service_account",
+                pgettext_lazy("Permission description", "Manage service account"),
+            ),
+        )
+
+    def _get_permissions(self) -> Set[str]:
+        """Return the permissions of the service."""
+        if not self.is_active:
+            return set()
+        perm_cache_name = "_service_perm_cache"
+        if not hasattr(self, perm_cache_name):
+            perms = self.permissions.all()
+            perms = perms.values_list("content_type__app_label", "codename").order_by()
```

Comment:
Is the empty `order_by` accidental?

Reply:
To keep consistency between User and ServiceAccount I took permission logic from django.
`django.contrib.auth.backends.ModelBackend#_get_permissions` Here you can find original implementation. I've just changed some part of the flow.